jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -241,7 +241,14 @@ MethodHandle cloneWithNewType(MethodType newType) {
 	/*[IF ]*/
 	// Until the JIT can synthesize calls to virtual methods, we must synthesize calls to these static ones instead
 	/*[ENDIF]*/
-	private static MethodHandle asType(MethodHandle mh, MethodType newType) { return mh.asType(newType); }
+	private static MethodHandle asType(MethodHandle mh, MethodType newType) {
+		/* JIT can easily propagate type information and fold the if when it can prove early return always happen */
+		/* The early return also saves the JIT from having to inline full asType call */
```

Comment:
Please merge this into a single block comment:
```
/*
 *
 */
```

Reply:
Merged.